### DISCUSSION TITLE
Why doesn't the  /chat-messages API allow passes context messages as a list?

### DISCUSSION THREAD

**Discussion by jamie0725** [langgenius/dify#3869] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Pleas do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing?

Hi, I am new to the platform, but while I look through the API documentation (as attached below), I notice that the user is not allowed to pass the context history on his own. The feature is important for customer support scenarios where the AI agent is not the only responder in the conversation (i.e., there might be a human agent and other business messages previously involved). This means that relying solely on the context history of the agent will not suffice to answer the user's question:
```
curl -X POST 'https://api.dify.ai/v1/chat-messages' \
--header 'Authorization: Bearer {api_key}' \
--header 'Content-Type: application/json' \
--data-raw '{
    "inputs": {},
    "query": "What are the specs of the iPhone 13 Pro Max?",
    "response_mode": "streaming",
    "conversation_id": "",
    "user": "abc-123",
    "files": [
      {
        "type": "image",
        "transfer_method": "remote_url"
        "url": "https://cloud.dify.ai/logo/logo-site.png"
      }
    ]
}'
```

### 2. Describe the feature you'd like to see

I'd like to have the `query` field to support list conversations to carry the context (similar to ChatGPT [messages](https://platform.openai.com/docs/api-reference/chat/create#chat-create-messages)),  and in this case not using the context information of the dify platform.
<img width="644" alt="image" src="https://github.com/langgenius/dify/assets/27820994/e7227f95-75dc-454d-b86d-9e34338c2a3b">

### 3. How will this feature improve your workflow or experience?

a must for customer support, if no workaround is available.

### 5. Can you help us with this feature?

- [ ] I am interested in contributing to this feature.

**Comment by crazywoola** [comment]
Dify provide apis to the whole application. It's not a proxy to other api like OpenAI itself.

**Reply by jamie0725** [reply]
Yes I understand, but looking at my scenario (which could be quite common), I still think that allowing the user to pass in the context history (at least the user and assistant messages) would be an important feature.

**Reply by jamie0725** [reply]
Or does the context module have an API that allows the user to modify, which could be a workaround.

**Reply by MrOrz** [reply]
Despite the endpoint name being `/chat-messages`, the way this API works is more similar to OpenAI Assistant API, which manages the history of a thread all by itself and does not allow editing of the history.

A question that may arise after introducing the feature that specifies or modifies history would be: what should "Logs and Annotations" tab show after the user provides a new chat history? Having a good answer to that question may be a good start towards the actual implementation of such feature.

**Reply by glacierck** [reply]
> Despite the endpoint name being `/chat-messages`, the way this API works is more similar to OpenAI Assistant API, which manages the history of a thread all by itself and does not allow editing of the history.
> 
> A question that may arise after introducing the feature that specifies or modifies history would be: what should "Logs and Annotations" tab show after the user provides a new chat history? Having a good answer to that question may be a good start towards the actual implementation of such feature.

`/chat-messages` no longer supports modifying historical messages!They canceled 'parent_message_id'!!!

**Comment by f1025211271** [comment]
I encountered the same problem. When using the service api v1/chat-messages, there is no "parent_message_id" field, so I cannot ask a question again based on a historical message. I sincerely hope that this function will be added as soon as possible.

**Comment by varunmehra5** [comment]
Following this thread.

**Comment by wdxtub** [comment]
I have the same problem.
